types/artifacts.d.ts

```diff
@@ -509,7 +509,7 @@ declare global {
       export interface MetricComputationDataInput {
         devtoolsLog: DevtoolsLog;
         trace: Trace;
-        settings: Config.Settings;
+        settings: Immutable<Config.Settings>;
```

Comment:
this is complicated because this `settings` is just as often not `LH.Config.Settings` as it is, but that's something to sort out on a different day :)